bot/cogs/help.py

```diff
@@ -68,499 +67,307 @@ class HelpSession:
 
-        self._handle_not_found(query)
+            # else aliases with parent if group. we need to strip() in case it's a Command and `full_parent` is None,
+            # otherwise we get 2 commands: ` help` and normal `help`.
+            # We could do case-by-case with f-string but this is the cleanest solution
+            choices.update(f"{c.full_parent_name} {a}".strip() for a in c.aliases)
 
-    def _handle_not_found(self, query: str) -> None:
+            # all cog names
+        choices.update(self.context.bot.cogs)
+
+        # all category names
+        choices.update(getattr(n, "category", None) for n in self.context.bot.cogs if hasattr(n, "category"))
+        return choices
+
+    def command_not_found(self, string: str) -> "HelpQueryNotFound":
         """
-        Handles when a query does not match a valid command or cog.
+        Handles when a query does not match a valid command, group, cog or category.
 
-        Will pass on possible close matches along with the `HelpQueryNotFound` exception.
+        Will return an instance of the `HelpQueryNotFound` exception with the error message and possible matches.
         """
-        # Combine command and cog names
-        choices = list(self._bot.all_commands) + list(self._bot.cogs)
+        choices = self.get_all_help_choices()
+        result = process.extractBests(string, choices, scorer=fuzz.ratio, score_cutoff=90)
 
-        result = process.extractBests(query, choices, scorer=fuzz.ratio, score_cutoff=90)
+        return HelpQueryNotFound(f'Query "{string}" not found.', dict(result))
 
-        raise HelpQueryNotFound(f'Query "{query}" not found.', dict(result))
+    def subcommand_not_found(self, command: Command, string: str) -> "HelpQueryNotFound":
+        """
+        Redirects the error to `command_not_found`.
 
-    async def timeout(self, seconds: int = 30) -> None:
-        """Waits for a set number of seconds, then stops the help session."""
-        await asyncio.sleep(seconds)
-        await self.stop()
+        `command_not_found` deals with searching and getting best choices for both commands and subcommands.
+        """
+        return self.command_not_found(f"{command.qualified_name} {string}")
 
-    def reset_timeout(self) -> None:
-        """Cancels the original timeout task and sets it again from the start."""
-        # cancel original if it exists
-        if self._timeout_task:
-            if not self._timeout_task.cancelled():
-                self._timeout_task.cancel()
+    async def send_error_message(self, error: HelpQueryNotFound) -> None:
+        """Send the error message to the channel."""
+        embed = Embed(colour=Colour.red(), title=str(error))
 
-        # recreate the timeout task
-        self._timeout_task = self._bot.loop.create_task(self.timeout())
+        if getattr(error, "possible_matches", None):
+            matches = "\n".join(f"`{n}`" for n in error.possible_matches.keys())
```

Comment:
`.keys()` is redundant,
```suggestion
            matches = "\n".join(f"`{n}`" for n in error.possible_matches)
```